packages/edit-site/src/components/sidebar-navigation-screen-page/status-label.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { dateI18n, getDate, getSettings, humanTimeDiff } from '@wordpress/date';
+import { createInterpolateElement } from '@wordpress/element';
+import { Path, SVG } from '@wordpress/primitives';
+
+const publishedIcon = (
+	<SVG fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+		<Path
+			fillRule="evenodd"
+			clipRule="evenodd"
+			d="M8 16A8 8 0 1 0 8 0a8 8 0 0 0 0 16Zm-1.067-5.6 4.2-5.667.8.6-4.8 6.467-3-2.267.6-.8 2.2 1.667Z"
+		/>
+	</SVG>
+);
+
+const draftIcon = (
+	<SVG fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+		<Path
+			fillRule="evenodd"
+			clipRule="evenodd"
+			d="M14.5 8a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0ZM16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0Zm-8 4a4 4 0 0 0 0-8v8Z"
+		/>
+	</SVG>
+);
+
+const pendingIcon = (
+	<SVG fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+		<Path
+			fillRule="evenodd"
+			clipRule="evenodd"
+			d="M14.5 8a6.5 6.5 0 1 1-13 0 6.5 6.5 0 0 1 13 0ZM16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0ZM8 4a4 4 0 1 0 0 8 4 4 0 0 0 0-8Z"
+		/>
+	</SVG>
+);
+
+export default function StatusLabel( { status, date } ) {
+	const relateToNow = humanTimeDiff( date );
+	let statusLabel = '';
+	let statusIcon = pendingIcon;
+	switch ( status ) {
+		case 'publish':
+			statusLabel = createInterpolateElement(
+				sprintf(
+					/* translators: %s: is the relative time when the post was published. */
+					__( 'Published <time>%s</time>' ),
+					relateToNow
+				),
+				{ time: <time dateTime={ date } /> }
+			);
+			statusIcon = publishedIcon;
+			break;
+		case 'future':
+			const formattedDate = dateI18n(
+				getSettings().formats.date,
+				getDate( date )
+			);
+			statusLabel = createInterpolateElement(
+				sprintf(
+					/* translators: %s: is the formatted date and time on which the post is scheduled to be published. */
+					__( 'Scheduled for <time>%s</time>' ),
+					formattedDate
+				),
+				{ time: <time dateTime={ date } /> }
+			);
+			break;
+		case 'draft':
+			statusLabel = __( 'Draft' );
+			statusIcon = draftIcon;
+			break;
+		case 'pending':
+			statusLabel = __( 'Pending' );
+			break;
+	}
+
+	return (
+		<div
+			className={ classnames(
+				'edit-site-sidebar-navigation-screen-page__status',
+				{
+					[ `has-status has-${ status }-status` ]: !! status,
+				}
+			) }
+		>
+			{ statusIcon } { statusLabel }
+		</div>
+	);
+}
```

Comment:
I believe we should hide the whole row if we determine a status, thanks for confirming this. I'll update.

Reply:
Actually 'private' is a status, but we don't cater for it in terms of icon or copy

Reply:
Also, what about password protected? We can detect that as well. Perhaps we need more icons?  

I've done this for now:

<img width="323" alt="Screenshot 2023-05-26 at 2 02 46 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/ead32940-2fdb-4cc2-990d-6018b483a62d">


<img width="335" alt="Screenshot 2023-05-26 at 2 02 56 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/7d21cde6-9956-4511-8d26-4a508f35447d">

Reply:
Yeah, for private pots, I was imagining it could just be a `Private` status label with no icon, a little bit like the `pending` status for now.

But also, feel free to leave it for a follow-up — this PR is already fairly large, so it could be easier to fine-tune / fix up these things in follow-ups if you'd prefer. Whichever works best for you!

Reply:
Sorry, missed your reply before commenting. Those work nicely for me!

Reply:
Thanks! Icons are easy to add in a follow up. Let's merge this!